src/db/util.rs

```diff
@@ -0,0 +1,63 @@
+use std::fmt;
+use axum::{
+    async_trait,
+    extract::{
+        FromRef,
+        FromRequestParts
+    },
+    http::{
+        StatusCode,
+        request::Parts,
+    },
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use sqlx::{
+    Postgres,
+    pool::PoolConnection,
+    postgres::PgPool
+};
+use time::{
+    macros::format_description,
+    PrimitiveDateTime,
+};
+
+pub(crate) type DBResult<T, E = sqlx::Error> = std::result::Result<T, E>;
+
+pub(crate) struct DatabaseConnection(pub(crate) PoolConnection<Postgres>);
+
+#[derive(Deserialize, Debug, Clone, Serialize)]
+pub(crate) struct DbDate {
+    #[serde(skip_deserializing, skip_serializing_if = "Option::is_none")]
+    pub datetime: Option<PrimitiveDateTime>,
+}
+
+impl fmt::Display for DbDate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let format = format_description!("[year]-[month]-[day] [hour]:[minute]:[second]");
```

Comment:
This should be 

```suggestion
        let format = format_description!("[year]-[month]-[day]T[hour]:[minute]:[second]");
```

according to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#value